src/AsyncWebSynchronization.h

```diff
@@ -7,6 +7,36 @@
 
 #ifdef ESP32
 
+// This is the ESP32 version of the Sync Lock, using the FreeRTOS Semaphore
+// Modified 'AsyncWebLock' to just only use mutex since pxCurrentTCB is not
+// always available. According to example by Arjan Filius, changed name,
+// added unimplemented version for ESP8266
+class AsyncPlainLock
+{
+private:
+  SemaphoreHandle_t _lock;
+  mutable void *_lockedBy;
+
+public:
+  AsyncPlainLock() {
+    _lock = xSemaphoreCreateBinary();
+    xSemaphoreGive(_lock);
+  }
+
+  ~AsyncPlainLock() {
+    vSemaphoreDelete(_lock);
+  }
+
+  bool lock() const {
+      xSemaphoreTake(_lock, portMAX_DELAY);
```

Comment:
xSemaphoreTake() can fail

Reply:
No, actually in this case, xSemaphoreTake, which is an alias to xQueueGenericReceive, would only ever have a nonzero return value if an effective timeout was supplied as an argument. But the portMAX_DELAY in this case means the timeout is disabled. xSemaphoreTake does either lock indefinitely or return zero.

Reply:
xQueueGenericReceive, I meant it only returns "pdFALSE" when invoked with a non-infinite timeout.

Reply:
Yes, you are right. I expected a ```(void)xSemaphoreTake(...)```, but this may depend on the coding style.